Vault-4964: Update Managed Key documentation for AWS KMS (#14321)
hashicorp/vault: website/content/docs/enterprise/managed-keys.mdx

## Backend Support

Managed Keys were developed to support different types of external backends.

> At this time supported backends are PKCS#11 and AWS KMS, with the possibility of

Reviewer: ```suggestion
Currently, supported backends are PKCS#11 and AWS KMS. 
```

> future integrations being added.

Reviewer: ```suggestion
Support for additional integrations may be added in the future.
```
